File: InheritanceTest/results/InheritanceTest.go
package main

import "fmt"
type IPrinterBase interface {
    SomeBaseFunc() int 
}

type IPrinter interface {
    IPrinterBase
    PrintIt() 
}

type BasePrinter struct {
}

func NewBasePrinter() *BasePrinter {
    this := new(BasePrinter)
    return this
}

func (this *BasePrinter) GetValue() string {
    return "Base"
}

func (this *BasePrinter) PrintIt() {
    fmt.Println(fmt.Sprintf("BasePrinter: %v", this.GetValue()))
}

func (this *BasePrinter) SomeBaseFunc() int {
    return 42
}

type ChildPrinter struct {
    BasePrinter
}

func NewChildPrinter() *ChildPrinter {
    this := new(ChildPrinter)
    return this
}

func (this *ChildPrinter) GetValue() string {
    return "Child"
}

type TestClass struct {
}

func NewTestClass() *TestClass {
    this := new(TestClass)
    return this
}

func (this *TestClass) GetPrinter(name string) IPrinter {
    var tmp0 IPrinter
    if name == "child" {
      tmp0 = NewChildPrinter()
    } else {
      tmp0 = NewBasePrinter()
    }
    result := tmp0
    return result
}

func (this *TestClass) TestMethod() {
    basePrinter := this.GetPrinter("base")
    childPrinter := this.GetPrinter("child")
    basePrinter.PrintIt()
    childPrinter.PrintIt()
    fmt.Println(fmt.Sprintf("%v == %v", basePrinter.SomeBaseFunc(), childPrinter.SomeBaseFunc()))
}

func init() {
}

func main() {
    defer func() {
      if r := recover(); r != nil {
          fmt.Print("Exception: ", r)
      }
    }()

    c := NewTestClass()
    c.TestMethod();
}

File: InheritanceTest/results/InheritanceTest.java
interface IPrinterBase {
  Integer someBaseFunc() throws Exception;
}

interface IPrinter extends IPrinterBase {
  void printIt() throws Exception;
}

class BasePrinter implements IPrinter
 {
    public String getValue() throws Exception
    {
        return "Base";
    }
    
    public void printIt() throws Exception
    {
        System.out.println("BasePrinter: " + this.getValue());
    }
    
    public Integer someBaseFunc() throws Exception
    {
        return 42;
    }
}

class ChildPrinter extends BasePrinter
 {
    public String getValue() throws Exception
    {
        return "Child";
    }
}

class TestClass
 {
    public IPrinter getPrinter(String name) throws Exception
    {
        IPrinter result = name.equals("child") ? new ChildPrinter() : new BasePrinter();
        return result;
    }
    
    public void testMethod() throws Exception
    {
        IPrinter basePrinter = this.getPrinter("base");
        IPrinter childPrinter = this.getPrinter("child");
        basePrinter.printIt();
        childPrinter.printIt();
        System.out.println(basePrinter.someBaseFunc() + " == " + childPrinter.someBaseFunc());
    }
}

class Program {
    public static void main(String[] args) throws Exception {
        try {
            new TestClass().testMethod();
        } catch (Exception err) {
            System.out.println("Exception: " + err.getMessage());
        }
    }
}